lib/withAuth.ts

```diff
@@ -1,10 +1,12 @@
 import type { NextApiRequest, NextApiResponse } from 'next';
+
 import User from '../models/db/user';
 import { UserModel } from '../models/mongoose';
 import dbConnect from './dbConnect';
 import jwt from 'jsonwebtoken';
 
 export type NextApiRequestWithAuth = NextApiRequest & {
+  current_user: User;
```

Comment:
nit: rename to just `user` to match userId
could also potentially just remove userId and update all `req.userId` references to `req.user._id`

Reply:
My thought process was that making it user may be confusing - especially if you are dealing with different users for different parts of the code that deals with other types of users.

current_user is also what Ruby on Rails uses, so seemed like a decent convention.

However, I guess since one would be writing req.user it would be clear that it is the request's user that you are referencing. So maybe it is ok